src/Equinor.ProCoSys.IPO.Command/Email/SmtpService.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System.Net.Mail;
+using System.Net;
+using System;
+using System.Threading.Tasks;
+using Equinor.ProCoSys.IPO.Command.InvitationCommands;
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.InvitationAggregate;
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.PersonAggregate;
+using System.IO;
+using System.Net.Mime;
+using System.Text;
+using Attachment = System.Net.Mail.Attachment;
+using System.Linq;
+using Equinor.ProCoSys.IPO.Command.InvitationCommands.CreateInvitation;
+using Equinor.ProCoSys.Auth.Authentication;
+using Microsoft.Extensions.Options;
+using Equinor.ProCoSys.Auth.Client;
+
+namespace Equinor.ProCoSys.IPO.Command.Email
+{
+    public class SmtpService : ISmtpService
+    {
+        protected readonly IOptionsMonitor<SmtpOptions> _smtpOptions;
+
+        public SmtpService(IOptionsMonitor<SmtpOptions> smtpOptions) => _smtpOptions = smtpOptions;
+
+        public async Task SendAsync(MailMessage message)
+        {
+            // SendMailAsync is not thread-safe, so to ensure it is not busy sending an mail I instantiate an new SmtpClient
+            var client = new SmtpClient(_smtpOptions.CurrentValue.Server, _smtpOptions.CurrentValue.Port);
```

Comment:
objedct initializer can be used here too:
![image](https://github.com/equinor/procosys-invitation-for-punch-out-api/assets/19545528/cd3e74b5-d63a-4e04-992f-deda756f8fd3)

Reply:
Outdated.